Add Map support for Locative component

danijelst opened the pull request:
**Description:**
The Locative App on the mobile is sending an HTTP request to the
server where also the GPS location is sent.
But the GPS location was not passed to the event device_tracker.see.

Use the passed GPS location from Locative and pass it to the
device_tracker.see event.

With this the device is then also shown on the HA Map component.

Commit:
Add Map support for Locative component

The Locative App on the mobile is sending an HTTP request to the
server where also the GPS location is sent.
But the GPS location was not passed to the event device_tracker.see.

Use the passed GPS location from Locative and pass it to the
device_tracker.see event.

With this the device is then also shown on the HA Map component.

Danielhiversen commented:
Great.
You have some failing tests, that needs to be fixed

kellerza commented on homeassistant/components/device_tracker/locative.py:
```diff
@@ -76,21 +76,24 @@ def _handle(self, data):
+        kwargs = {
+           'dev_id': device,
+           'location_name': location_name,
+           'gps': (data['latitude'], data['longitude'])
```
Can you please import some of these constants from `homeassistant.consts`? Like `ATTR_LATITUDE` and `ATTR_LONGITUDE`

Commit:
Use existing constants for latitude and longitude

Use the existing constants from homeassistant.consts:

ATTR_LATITUDE for 'latitude'
ATTR_LONGITUDE for 'longitude'

Danielhiversen commented on homeassistant/components/device_tracker/locative.py:
```diff
@@ -76,21 +77,24 @@ def _handle(self, data):
-            yield from self.hass.loop.run_in_executor(
-                None, partial(self.see, dev_id=device,
-                              location_name=location_name))
+            self.see(**kwargs)
```
I think you still have to use "yield from"
https://travis-ci.org/home-assistant/home-assistant/jobs/172768411#L1704

Commit:
Reuse the "yield from self.hass.loop.run_in_executor" again

Danielhiversen commented on homeassistant/components/device_tracker/locative.py:
```diff
@@ -76,21 +78,25 @@ def _handle(self, data):
                 None, partial(self.see, dev_id=device,
-                              location_name=location_name))
+                              location_name=location_name,
+                              gps=(data[ATTR_LATITUDE], data[ATTR_LONGITUDE])))
```
Why not use the gps_location variable? Otherwise you can remove the gps_location variable.

Commit:
Use variable gps_location

Danielhiversen commented:
Looks good to me
